sentry/src/routes/analytics.rs

```diff
@@ -0,0 +1,236 @@
+use crate::success_response;
+use crate::Application;
+use crate::ResponseError;
+use crate::RouteParams;
+use crate::Session;
+use bb8_postgres::tokio_postgres::Row;
+use chrono::Utc;
+use hyper::{Body, Request, Response};
+use primitives::adapter::Adapter;
+use redis::aio::MultiplexedConnection;
+use serde::{Deserialize, Serialize};
+use std::cmp;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct AnalyticsResponse {
+    time: u32,
+    value: String,
+}
+
+impl From<&Row> for AnalyticsResponse {
+    fn from(row: &Row) -> Self {
+        Self {
+            time: row.get("time"),
+            value: row.get("value"),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct AnalyticsQuery {
+    #[serde(default = "default_limit")]
+    pub limit: u32,
+    #[serde(default = "default_event_type")]
+    pub event_type: String,
+    #[serde(default = "default_metric")]
+    pub metric: String,
+    #[serde(default = "default_timeframe")]
+    pub timeframe: String,
+}
+
+impl AnalyticsQuery {
+    pub fn is_valid(&self) -> Result<(), ResponseError> {
+        let valid_event_types = ["IMPRESSION"];
+        let valid_metric = ["eventPayouts", "eventCounts"];
+        let valid_timeframe = ["year", "month", "week", "day", "hour"];
+
+        if !valid_event_types.iter().any(|e| *e == &self.event_type[..]) {
```

Comment:
you can just compare `String`s, why so complicated?
Same applies for the checks bellow.

Reply:
It wouldn't work because it would be comparing &str with String. I made the change to `.contains` though